### ISSUE TITLE
[bug] erroneous type errors

### ISSUE THREAD

**Issue by trusktr** [TypeStrong/atom-typescript#1487]
Here's a screenshot, notice that the tooltip shows a type error that has nothing to do with the line where it is showing the error:

![screen shot 2019-01-16 at 3 16 05 pm](https://user-images.githubusercontent.com/297678/51284840-ce1dcf80-19a1-11e9-946a-7d0305452b64.png)

I don't know how to reproduce it. It just happens sometimes. The only way to recover is `control`+`option`+`command`+`l` to restart.

**Comment by trusktr** [comment]
The `keyof` in the screenshot is referring to the one in the lines above.

**Comment by trusktr** [comment]
Alright, now looks like I've reached an unrecoverable state. No matter how many times I restart Atom, I am no longer seeing any time information or errors at all, in the entire project. I haven't touched any configs, I've been editing only a single source file (the one in the screenshot above).

**Comment by trusktr** [comment]
Alright, strange, finally it shows something, and it seems to indicate I now have a ton of type errors all over my source files (which is wrong):

![screen shot 2019-01-16 at 3 46 42 pm](https://user-images.githubusercontent.com/297678/51286012-ff989a00-19a5-11e9-9690-c63ca28b1c32.png)

**Comment by trusktr** [comment]
Don't know if it's related, but I'm seeing an errors like the following occurring in the console:

![screen shot 2019-01-16 at 3 48 22 pm](https://user-images.githubusercontent.com/297678/51286105-67e77b80-19a6-11e9-8e4e-abdcd69a53f3.png)

continued:

![screen shot 2019-01-16 at 3 48 49 pm](https://user-images.githubusercontent.com/297678/51286111-6d44c600-19a6-11e9-95c1-b618ada9fc41.png)

continued:

![screen shot 2019-01-16 at 3 49 04 pm](https://user-images.githubusercontent.com/297678/51286116-7170e380-19a6-11e9-99ed-b24c583fcdb0.png)

**Comment by trusktr** [comment]
My typescript experience is totally borked. Restarted Atom completely (quit all window, kill Atom) and still the same issues: erroneous errors, etc.

**Comment by trusktr** [comment]
Looks like 

```ts
type PointGroups = {
  [key in keyof Pick<Layers, 'base1' | 'base1color'>]: THREE.Group
}
```

was the source of all problems. When I changed it to

```ts
type PointGroups = {
  [key in keyof Pick<typeof Layers, 'base1' | 'base1color'>]: THREE.Group
}
```

with the insertion of the `typeof`, everything went back to normal.

**Comment by trusktr** [comment]
I feel that `declaration: true` is broken anyways. For example, [this issue](https://github.com/Microsoft/TypeScript/issues/17293) (there's other ones too).

**Comment by trusktr** [comment]
Looks to be an edge case that atom-typescript (or the ts server?) isn't handling well when I'm incorrectly passing the `const Layers` into `Pick` which expects a `type` and hence `typeof Layers` worked.

**Comment by lierdakil** [comment]
Thanks for the report.

Long story short you've apparently run into https://github.com/Microsoft/TypeScript/issues/25456, which is obviously upstream bug.

Perhaps tsserver errors should probably be reported more visibly to avoid unnecessary confusion? I'll see what I can do with that.
